tests/models/imagegpt/test_modeling_imagegpt.py

```diff
@@ -257,11 +257,9 @@ def _check_scores(self, batch_size, scores, length, config):
         self.assertEqual(len(scores), length)
         self.assertListEqual([iter_scores.shape for iter_scores in scores], [expected_shape] * len(scores))
 
-    @unittest.skip(
-        reason="After #33632, this test still passes, but many subsequential tests fail with `device-side assert triggered`"
-    )
+    @run_test_using_subprocess
     def test_beam_search_generate_dict_outputs_use_cache(self):
```

Comment:
Maybe we can mark as subprocess instead, and have a separate command to run these no?

Reply:
The idea here is not to to run these tests (very very few tests) separated from the other tests, but to change the behavior of its execution logic (run in a subprocess) even if a suite of tests is collected by `pytest`.

If we use separate command to run those (very few) tests, it means we need to add new jobs (or steps), and will get new separated test reports which is probably making the testing statistic less unified.

Reply:
Okay, this would have fixed the issue mentioned by Lysandre, and we don't need a new job, maybe we can run them after the normal ones?